test_app/tests/activitystream/test_api.py

```diff
@@ -97,3 +100,164 @@ def test_activitystream_api_filtering(admin_api_client, user):
     response = admin_api_client.get(url + '?' + urlencode(query_params))
     assert response.status_code == 200
     assert response.data['count'] == 1
+
+
+@pytest.mark.xfail(reason="TODO: Decide if we really want to exclude these from list view.")
```

Comment:
Should we make this decision now or is it ok to do that in the future?

Reply:
I'd like to hear @AlanCoding's thoughts. I think the concern was adding a bunch of queries. But with how I've done this, I don't see any more queries (according to django-debug-toolbar) than we currently get in devel when I go to /api/v1/activitystream.

So my inclination is that's fine. But I trust @AlanCoding's instinct about this, so if he still thinks it's an issue then I'd say we shouldn't do it on the list view.

If we do it on the list view, we're kinda stuck with that, because we can't pull fields from the API response without risking breaking people. That said, I kind of like having it on the list view and keeping the list view and the singular entry response consistent.

Reply:
Your use of caching is much more advanced than I anticipated. I tip my hat to @relrod 

If it feels like the methods you used in this PR are maintainable, then it would be fine to put them in the list view.